.github/workflows/buildAndDeploy.yml

```diff
@@ -18,17 +18,89 @@ jobs:
         if: github.event_name != 'pull_request'
         with:
           submodules: true
-      - name: Set up java 1.8
-        uses: actions/setup-java@v1
+      - name: Login to GitHub Package Registry
+        run: echo "${{ secrets.GITHUB_TOKEN }}" | docker login docker.pkg.github.com -u "${{ github.actor }}" --password-stdin
+      - name: Setup Buildx
+        id: buildx
+        uses: crazy-max/ghaction-docker-buildx@v3
+      - name: Cache Docker layers
+        uses: actions/cache@v2
+        id: cache
         with:
-          java-version: 1.8
+          path: /tmp/.buildx-cache
+          key: ${{ runner.os }}-buildx-${{ github.sha }}
+          restore-keys: |
+            ${{ runner.os }}-buildx-
       - name: Build docker image
-        if: github.event_name != 'pull_request' || github.event.pull_request.head.repo.full_name == 'datastax/cass-config-builder'
-        env:
-          ORG_GRADLE_PROJECT_dockerRegistry: docker.pkg.github.com
-          ORG_GRADLE_PROJECT_dockerRegistryUsername: ${{ github.actor }}
-          ORG_GRADLE_PROJECT_dockerRegistryPassword: ${{ secrets.GITHUB_TOKEN }}
-        uses: eskatos/gradle-command-action@v1
-        with:
-          gradle-version: 6.0.1
-          arguments: pushDockerImages
+        run: |
+          GITHUB_REPO_URL="https://github.com/${{ github.repository }}"
```

Comment:
What follows is me pulling all the docker stuff out of gradle and putting it here. There are a number of reasons for this:

- We need the config builder, management api, and the operator to make everything work. Each uses their own build system: gralde, mage, and maven. Implementing the same logic in three different ways isn't great.
- The gradle plugin appears to use the docker-java library under the hood, and it doesn't support buildkit or buildx (as discovered when adding arm64 support to the management api).
- In my humble opinion, this is just plain easier to understand.

Reply:
Looks great.  This is a good move.